app/controllers/faculty_access_controller.rb

```diff
@@ -38,4 +40,9 @@ def instructor_has_selected_subject(key)
     params[:apply] && params[:apply][:subjects] && params[:apply][:subjects][key] == '1'
   end
 
+  def disallow_signup
+    # value doesn't really matter, if set at all, signup hidden
+    params[:no_signup] = 1
+  end
```

Comment:
I kinda feel like it's bad form to update params, since they should always correspond to what you see in the URL and it's a bit confusing when they don't match.  Would it be better to set a `@no_signup` flag, and check that in the view?

Maybe that's also set somewhere else though and you're just forcing it to be set here?